packages/headless/src/features/question-answering/question-answering-slice.test.ts

```diff
@@ -199,4 +274,58 @@ describe('question answering slice', () => {
 
     expect(finalState.liked).toBeFalsy();
   });
+
+  it('when executeSearch is triggered again, resets only the altered snippets', () => {
+    const relatedQuestionsCount = 5;
+    const firstSearchResponse = buildUniqueSearchResponse(
+      0,
+      relatedQuestionsCount
+    );
+    const firstResponseState = questionAnsweringReducer(
+      state,
+      executeSearch.fulfilled(
+        buildMockSearch({response: firstSearchResponse}),
+        '',
+        null as never
+      )
+    );
+
+    const stateWithAllSnippetsExpanded =
+      firstResponseState.relatedQuestions.reduce(
+        (state, {questionAnswerId}) =>
+          questionAnsweringReducer(
+            state,
+            expandSmartSnippetRelatedQuestion({
+              questionAnswerId: questionAnswerId,
+            })
+          ),
+        firstResponseState
+      );
```

Comment:
expanding all snippets by calling the reducer multiple times in a reduce function 🤡